src/main/java/seedu/address/model/lesson/TimeSlot.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.lesson;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Represents the time in which a lesson takes place in the LessonBook.
+ */
+public class TimeSlot {
```

Comment:
Perhaps you could make `TimeSlot` an abstract class so that we can extend with `TemporaryTimeSlot` and `RecurringTimeSlot`. 😢